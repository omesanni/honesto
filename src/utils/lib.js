import React from 'react';
import AsyncComponent from '../components/AsyncComponent';

/**
 * Lazy loads a component
 * @param  {Function} getComponent Returns a dynamic import of the component
 * @param  {Object}   props
 * @return {ReactElement}
 */
export function lazyLoadComponent(getComponent, props = {}) {
  return React.createElement(AsyncComponent, { ...props, render: getComponent });
}
