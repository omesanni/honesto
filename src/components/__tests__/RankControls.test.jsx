import React from 'react';
import { shallow } from 'enzyme';
import RankControls from '../RankControls';
import { expectToMatchSnapshot } from '../../utils/tests';

describe('RankControls', () => {
  let component;

  const props = {
    score: 20,
    onUpvote: jest.fn(),
    onDownvote: jest.fn(),
  };

  function getRankControls() {
    return <RankControls {...props} />;
  }

  function mountComponent() {
    return shallow(getRankControls());
  }

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders', () => {
    expectToMatchSnapshot(getRankControls());
  });

  it('should upvote', () => {
    component = mountComponent();
    component.find('.rank-controls__up').simulate('click');

    expect(props.onUpvote).toHaveBeenCalled();
  });

  it('should downvote', () => {
    component = mountComponent();
    component.find('.rank-controls__down').simulate('click');

    expect(props.onDownvote).toHaveBeenCalled();
  });
});
