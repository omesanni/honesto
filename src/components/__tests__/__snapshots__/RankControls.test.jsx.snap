// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankControls renders 1`] = `
<div
  className="rank-controls"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-up fa-w-14 rank-controls__up mr-3"
    data-icon="arrow-up"
    data-prefix="fas"
    onClick={[MockFunction]}
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <span
    className="badge badge-pill badge-secondary"
  >
    20
  </span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-down fa-w-14 rank-controls__down ml-3"
    data-icon="arrow-down"
    data-prefix="fas"
    onClick={[MockFunction]}
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
